### PR TITLE
added ability to set application name

### PR DESCRIPTION
**Pull request by jhinds**

### CODE REVIEW

**Comment by jhinds** [comment]
added a level for CRITCAL

**Comment by stewrutledge** [comment]
One concern I have with this change is that self.application will be reported as None if none is given, which will generate some misleading log messages.

Maybe something like:

```python
if self.application:
    msgDict['application'] = self.application
```

How are log messages with using `kw.get('application', None)` represented in for example logstash?

**Comment by stewrutledge** [comment]
Also could you redo the commit for adding the level Critical?

You've included the file output from `python setup.py build`

**Comment by jhinds** [comment]
I'll implement your suggestion for logging an application name if there is none, that makes sense.

Could you rephrase or be more specific about your question at the end of the comment?

**Comment by stewrutledge** [comment]
I may have been imagining things! I thought your first commit including additional files generation by `python setup.py build` that don't need to be version controlled.

In any event, this looks good now. I'm going to give it a quick try and merge it in hopefully tomorrow and push a new version to pypi.

**Comment by stewrutledge** [comment]
Merged
